openbook_auth/models.py

```diff
@@ -599,11 +600,26 @@ def get_comments_for_post_with_id(self, post_id, max_id=None):
         if not Post.post_with_id_has_public_comments(post_id):
             comments_query = Q(commenter_id=self.pk)
 
+        max_id_query_results = []
+        min_id_query_results = []
+        PostComment = get_post_comment_model()
         if max_id:
-            comments_query.add(Q(id__lt=max_id), Q.AND)
+            max_id_query_results = PostComment.objects.filter(comments_query
+                                                              & Q(id__lt=max_id)).order_by('-created')[:count]
```

Comment:
Counting and ordering are operations done on a resulting queryset, never on a method that should produce the queryset. Move this operations outside, where this function is called.

Reply:
I will need some explanation as why do we even need to do this, it's just an extra `min_id` parameter here. the rest should be handled on the view.

Reply:
And it shouldnt be max id and in id. You don't need both. It's either one or the other. You won't ever load more posts up and down on the same call. It will always be one directional scrolling